scripts/vehicle_node.py

```diff
@@ -0,0 +1,292 @@
+        self.pose.position.y += (distance)*sin(self.vehicle_yaw)
+
+        #Calculate z position using linear interpolation and create cloud array
+        
+        #1. Create range to be used in interpolation function
+        x = np.arange(0, self.gaussian_array.shape[0]*self.resolution, self.resolution)
+        y = np.arange(0, self.gaussian_array.shape[1]*self.resolution, self.resolution)
+
+        #2. Create cloud array to be converted to point cloud for vizualization
+        #TODO do this without for loops
+        cloud = []
+        for i in range(self.gaussian_array.shape[0]):
+            for j in range(self.gaussian_array.shape[1]):
+                innerlist = []
+                innerlist.append(x[i])
+                innerlist.append(y[j])
+                innerlist.append(self.gaussian_array[i][j])
+                cloud.append(innerlist)
+
+        #3. Create interpolation function based on the ranges and gaussian data
+        f = RectBivariateSpline(x,y, self.gaussian_array)
+
+        #4. Find z value for x and y coordinate of vehicle using interpolation function
+        self.pose.position.z = f(self.pose.position.x, self.pose.position.y)
+
+        #Convert Euler Angles to Quarternion
+        q = tf.transformations.quaternion_from_euler(0.0, 0.0, self.vehicle_yaw)
+
+        #Broadcast vehicle frame which is a child of the world frame
+        br = tf.TransformBroadcaster()
+        br.sendTransform((self.pose.position.x, self.pose.position.y, self.pose.position.z), 
+                        q, rospy.Time.now(),"vehicle_frame", "map")
+
+        #Construct the homogenous transformation matrix for map to vehicle frame
+        translation = [self.pose.position.x, self.pose.position.y, self.pose.position.z]
+        map_t_vehicle = tf.transformations.translation_matrix(translation)
+        map_R_vehicle   = tf.transformations.quaternion_matrix(q)
+        map_T_vehicle = np.zeros((4,4))
+        map_T_vehicle[0:3,0:3] = map_R_vehicle[0:3,0:3]
+        map_T_vehicle[:4,3] = map_t_vehicle[:4,3]
+
+        #Create footprint of vehicle
+        x1 = np.linspace((-self.vehicle_length), (self.vehicle_length),30)
+        y1 = np.linspace((-self.vehicle_width), (self.vehicle_width),15)
+        x1, y1 = np.meshgrid(x1, y1)
+        x1 = x1.ravel()
+        y1 = y1.ravel()
+
+        #For every point in the vehicle footprint, calculate the position wrt to the vehicle's frame
+        # and its interpolated z value. Add this point to a list of points for visualization.
+        points = []
+        for i in range(x1.shape[0]):
+            p = Point()
+            footprint = np.array([[x1[i]],[y1[i]],[0.0],[1.0]])
+            footprint = np.matmul(map_T_vehicle, footprint)
+            footprint[2,0] =  f(footprint[0,0], footprint[1,0])
+            p.x = footprint[0,0]
+            p.y = footprint[1,0]
+            p.z = footprint[2,0]
+            points.append(p)
```

Comment:
You can flatten most of this into a single matrix multiply.  No need for the for loop.  This isn't a blocker for merging, but at least put a TODO.